src/averageAges.js

```diff
@@ -37,7 +41,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren === undefined
+    ? people.filter(woman => woman.sex === 'f')
+
+    : people.filter(woman => people.some(child => child.mother === woman.name));
+
+  return women
+    .map(woman => woman.died - woman.born)
+    .reduce((prev, cur) => prev + cur) / women.length;
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.

Also as an option(up to you) you could create 2 separate variables: `women` and `womenWithChildren` and pass them as arguments to call your new `getAverage` function. This way ternary operator would be in return statement and code would be more clear.

Reply:
don`t understand about 2 separate variables